python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -246,6 +246,14 @@ def is_empty(self) -> bool:
     def empty() -> "TimeWindow":
         return TimeWindow(start=datetime.max, end=datetime.max)
 
+    @lru_cache(maxsize=1000)
+    def start_timestamp(self) -> float:
+        return self.start.timestamp()
+
+    @lru_cache(maxsize=1000)
+    def end_timestamp(self) -> float:
+        return self.end.timestamp()
```

Comment:
can you use `memray` on the repro case you have? given these are floats I suspect we could `maxsize=None` these

Reply:
The concern would be memory growing unboundedly gradually over time right? Does doing it on the repro case really help with that?